packages/auth/ios/RNFBAuth/RNFBAuthModule.m

```diff
@@ -1667,7 +1667,8 @@ - (NSDictionary *)firebaseUserToDict:(FIRUser *)user {
       @"uid" : hint.UID,
       @"factorId" : hint.factorID == nil ? [NSNull null] : [self getJSFactorId:(hint.factorID)],
       @"displayName" : hint.displayName == nil ? [NSNull null] : hint.displayName,
-      @"enrollmentDate" : enrollmentDate,
+      @"enrollmentTime" : enrollmentDate,
+      @"phoneNumber" : hint.phoneNumber
```

Comment:
the if-null / do-this / else do that above this line are from previous crashes, seems better to be safe than sorry by using the same idiom as above

```suggestion
      @"phoneNumber" : hint.phoneNumber == nil ? [NSNull null] : hint.phoneNumber
```

Reply:
The documentation seems to suggest otherwise to me, https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRPhoneMultiFactorInfo - `phoneNumber` is marked as non-null

I've read through the other PR https://github.com/invertase/react-native-firebase/pull/7541 regarding `factorId` being and I'm surprised this was needed. Feels like a bug in the underlying firebase SDK to me - with a null `factorId` you won't be able to tell whether to follow the phone or totp flows. Though given the auth part isn't opensource I guess there's little hope of debugging/fixing it upstream 😢

Happy to add the extra defensiveness here, but I think ideally it wouldn't be required.

Reply:
Though as per https://github.com/invertase/react-native-firebase/pull/7565/files#r1451696979 I'm not so hot on objective c - if it will allow a `FIRMultiFactorInfo` where we've said we're looping over `FIRPhoneMultiFactorInfo` then this definitely needs a nil check, as a TOTP factor will be a `FIRMultiFactorInfo`

Reply:
`nil` check added to account for `FIRMultiFactorInfo`, and also added an assertion in the e2e tests (though unfortunately it looks like these are skipped for [ios](https://github.com/invertase/react-native-firebase/blob/8b4e39baddec2f9ac0569489d47b8a3915056ef7/packages/auth/e2e/multiFactor.e2e.js#L546-L548))